lib/style/deprecations.ex

```diff
@@ -0,0 +1,64 @@
+# Copyright 2024 Adobe. All rights reserved.
+# This file is licensed to you under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License. You may obtain a copy
+# of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+# OF ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+defmodule Styler.Style.Deprecations do
+  @moduledoc """
+  Transformations to soft or hard deprecations introduced on newer Elixir releases
+  """
+
+  @behaviour Styler.Style
+
+  def run({node, meta}, ctx), do: {:cont, {style(node), meta}, ctx}
+
+  # Logger.warn => Logger.warning
+  # Started to emit warning after Elixir 1.15.0
+  defp style({{:., dm, [{:__aliases__, am, [:Logger]}, :warn]}, funm, args}),
+    do: {{:., dm, [{:__aliases__, am, [:Logger]}, :warning]}, funm, args}
+
+  # Path.safe_relative_to/2 => Path.safe_relative/2 
+  # Path.safe_relative/2 is available since v1.14
+  # TODO: Remove after Elixir v1.19
```

Comment:
could do `if Version.match?(System.version(), "< 1.19.0-dev"))` and not have a todo eh :)

Reply:
I don't know if we should do that because the TODO comes from the Elixir source code, meaning we can't be sure if they will delete the code for that release.